scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -1661,19 +1692,14 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
 
   def caseClause(): Case = atPos(in.prevTokenPos, auto) {
     require(token.isNot[`case`] && debug(token))
-    token match {
-      case token: Unquote if token.prev.is[Ellipsis] =>
-        unquote[Case]
```

Comment:
But `case x => y; case q => w` is not case-separated. There is one extra `case`.